homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,206 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hive/
+"""
+import logging
+from datetime import datetime
+
+from homeassistant.components.climate import (ClimateDevice,
+                                              STATE_AUTO, STATE_HEAT,
+                                              STATE_OFF, STATE_ON)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+HIVE_TO_HASS_STATE = {'SCHEDULE': STATE_AUTO, 'MANUAL': STATE_HEAT,
+                      'ON': STATE_ON, 'OFF': STATE_OFF}
+HASS_TO_HIVE_STATE = {STATE_AUTO: 'SCHEDULE', STATE_HEAT: 'MANUAL',
+                      STATE_ON: 'ON', STATE_OFF: 'OFF'}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Hive climate devices."""
+    session = hass.data.get(DATA_HIVE)
+
+    add_devices([HiveClimateEntity(session, discovery_info)])
+
+
+class HiveClimateEntity(ClimateDevice):
+    """Hive Climate Device."""
+
+    def __init__(self, hivesession, hivedevice):
+        """Initialize the Climate device."""
+        self.node_id = hivedevice["Hive_NodeID"]
+        self.node_name = hivedevice["Hive_NodeName"]
+        self.device_type = hivedevice["HA_DeviceType"]
+        self.session = hivesession
+
+        if self.device_type == "Heating":
+            self.session.heating = self.session.core.Heating()
+            self.modes = [STATE_AUTO, STATE_HEAT, STATE_OFF]
+        elif self.device_type == "HotWater":
+            self.session.hotwater = self.session.core.Hotwater()
+            self.modes = [STATE_AUTO, STATE_ON, STATE_OFF]
+
+        self.session.entities.append(self)
+
+    def handle_update(self, updatesource):
+        """Handle the new update request."""
+        if '{}.{}'.format(self.device_type, self.node_id) not in updatesource:

```

Comment:
I don't understand what's happening here. Why should all hive entities except the one where the update originated from be scheduled for a state update?

Reply:
The entity that is not updated in this `handle_update` is the one that made the request for all entities to be updated. I excluded it as it already had the updated data as it was the one that made the call which resulted in new data

Reply:
But why should a change in state of one entity update the state of the other entities? Why should there be that connection?

Reply:
because, when you set a new temperature for example, the Hive api returns the latest data for all your Hive devices, so to me it makes sense to utilise as soon as possible that latest data by getting all the HA entities to refresh. They wont go out to the Hive api as it has only just been retrieved to HA cache.  It is the same for when you set temperature, or turn on a light, each time you do anything with the Hive API it will return the latest data for all devices.

It there any reason why it is not allowed to do this?

Reply:
No, I hadn't read all the previous discussions. As said in those, the dispatcher could be used for this case. But I think it's ok like this.

Reply:
ok, great, thanks.